aten/src/ATen/native/mkl/SpectralOps.cpp

```diff
@@ -61,7 +73,7 @@ void _fft_fill_with_conjugate_symmetry_slice(
   // We explicitly loop over one row, then use this lambda to iterate over
   // n-dimensions. This advances iter_index by one row, while updating in_ptr
   // and out_ptr to point to the new row of data.
-  auto advance_index = [&] {
+  auto advance_index = [&] () __ubsan_ignore_undefined__ {
```

Comment:
This shouldn't exhibit undefined behavior. Let's investigate some more. Maybe start by typing the loop variable correctly to `iter_indexes` size type. The loop is currently doing an unsigned vs signed comparison.

Reply:
>This shouldn't exhibit undefined behavior.

It doesn't, ubsan just doesn't seem to like this line:
https://github.com/pytorch/pytorch/blob/f9b5cd21d63950743668ff8eabf39171c8c7d54c/aten/src/ATen/native/mkl/SpectralOps.cpp#L88

because `out_strides` can be negative. It seems to be treating the pointer as if it were unsigned for some reason.

>Maybe start by typing the loop variable correctly to `iter_indexes` size type.

`DimVector::size_type` is just `size_t` which is already the type.

Reply:
My mistake, I thought it was int64_t.

Reply:
Either this is a false positive, or the the error message is just misleading. It specifically says unsigned offset, but `IntArrayRef` is `int64_t` so `out_strides` is definitely a signed offset.

Reply:
Let's assume the error message is misleading, but there's a valid UBSAN issue.

Could this be complaining, for example, that 

```
out_ptr += (signal_half_sizes[i] - 1) * out_strides[i];
```

is upsetting the sanitizer because it's applying a negative offset to out_ptr?

The message states that address 0x619001e07ca0 "overflows" to 0x619001e07c90. Let's just focus on those last two digits where these values are different. 0xa0 = 160, and 0x90 = 144. So Maybe the sanitizer thinks there was overflow because it's not expected a negative offset to be `+=` to the pointer to arrive at a final offset of 144?

Reply:
Yes, that's what I've been saying. The sanitizer doesn't seem to like that the calculated offset is negative. However, it's entirely expected to be negative because it's using negative strides. And as far as I'm aware, there's no undefined behaviour there.